Docs: Note when using For and No Data in alert rule (#19185)
grafana/grafana: docs/sources/alerting/rules.md

### For

> **Important note regarding No Data:**
>

> > Do not use `For` with the `If no data or all values are null` setting set to `No Data`. The triggering of `No Data` will trigger instantly and not take `For` into consideration. This may also result in that an OK notification not being sent if alert transitions from `No Data -> Pending -> OK`.

Reviewer: Is this a temporary notice? Shouldn't this be fixed in the next versions?

Author: Given that we're not sure if we want to add this support and/or how to implement it given next generation alerting is going to happen eventually and/or if we have time I think documenting this is needed until we've figured out what to do with this.

> > When configuring an alert to use `For` and `If no data or all values are null` set to `No Data` the triggering of `No Data` will trigger instantly and not take `For` into consideration. This may also result in that an OK notification not being sent if alert transitions from `No Data -> Pending -> OK`.

Reviewer 2: ```suggestion
> Do not use `For` with the `If no data or all values are null` setting set to `No Data`. The triggering of `No Data` will trigger instantly and not take `For` into consideration. This may also result in that an OK notification not being sent if alert transitions from `No Data -> Pending -> OK`.
```
